File: Java/2024_cursoemvideo_poo/aula08_agregacaoObjetos/src/Luta.java
import java.util.Objects;

public class Luta {
    private Lutador desafiado;
    private Lutador desafiante;
    private int rounds;
    private boolean aprovada;

    // Métodos Públicos
    public void marcarLuta(Lutador l1, Lutador l2) {
        if (l1 == l2) {
            this.aprovada = true;
            this.desafiado = l1;
            this.desafiante = l2;
        } else {
            System.out.println("A luta não pode acontecer.");
            if (!Objects.equals(l1, l2)) {
                System.out.println("Marcou o mesmo lutador 2 vezes.");
            } else {
                System.out.println("Os lutadores são de categorias diferentes");
                System.out.println("Lutador 1: " + l1.getNome() + "->Peso: " + l1.getCategoria());
                System.out.println("Lutador 1: " + l2.getNome() + "->Peso: " + l2.getCategoria());
            }
        }
    }

    public void luta() {

    }

    //Métodos Especiais
    public Luta(Lutador desafiado, Lutador desafiante, int rounds, boolean aprovada) {
        this.desafiado = desafiado;
        this.desafiante = desafiante;
        this.rounds = rounds;
        this.aprovada = aprovada;
    }

    public Lutador getDesafiado() {
        return desafiado;
    }

    public void setDesafiado(Lutador desafiado) {
        this.desafiado = desafiado;
    }

    public Lutador getDesafiante() {
        return desafiante;
    }

    public void setDesafiante(Lutador desafiante) {
        this.desafiante = desafiante;
    }

    public int getRounds() {
        return rounds;
    }

    public void setRounds(int rounds) {
        this.rounds = rounds;
    }

    public boolean isAprovada() {
        return aprovada;
    }

    public void setAprovada(boolean aprovada) {
        this.aprovada = aprovada;
    }
}

File: PHP/2016_cursoemvideo_poo/aula07e08_relacionamento/Luta.php
<?php 
require_once 'Lutador.php';

class Luta {
    private $desafiante, $desafiado, $rounds, $aprovada;
    
    function marcarLuta() {
        if ($this->getDesafiante() == $this->getDesafiado()) {
            echo "Erro: lutador foi inscrito duas vezes.<br><br>";
        } else {
            if ($this->getDesafiante()->getCategoria() == $this->getDesafiado()->getCategoria()) {
                $this->setAprovada(true);
                echo "Luta marcada. Perparem-se!<br><br>";
            }
            else {
                echo "Erro: lutadores de categorias diferentes!<br><br>";
            }
        }
    }

    function lutar() {
        if ($this->getAprovada() == false) {
            echo "Essa luta não foi aprovada.<br><br>";
        } else {
            $y1 = random_int(1,7);
            $y2 = random_int(0,2);
            
            if ($y2 == 0) {
                $this->setRounds($y1);
                $this->getDesafiante()->setResultado('vit');
                $this->getDesafiado()->setResultado('der');
                echo "Vitória para o lutador ". $this->getDesafiante()->getNome() ."!<br>". $this->getDesafiado()->getNome() ." foi derrotado.<br>";
            }
            elseif ($y2 == 1) {
                $this->setRounds($y1);
                $this->getDesafiante()->setResultado('der');
                $this->getDesafiado()->setResultado('vit');
                echo "Vitória para o lutador ". $this->getDesafiado()->getNome() ."!<br>". $this->getDesafiante()->getNome() ." foi derrotado.<br>";
            }
            else {
                $this->setRounds($y1);
                $this->getDesafiante()->setResultado('emp');
                $this->getDesafiado()->setResultado('emp');
                echo "Os lutadores ". $this->getDesafiante()->getNome() ." e ". $this->getDesafiado()->getNome() . " terminaram empatados.<br>";
            }
        }
    }

    public function __construct($desafiante, $desafiado) {
        $this->desafiante = $desafiante;
        $this->desafiado = $desafiado;
        $this->aprovada = false;
    }
    function getDesafiante() {return $this->desafiante;}
    function setDesafiante($desafiante) {$this->desafiante = $desafiante;}
    function getDesafiado() {return $this->desafiado;}
    function setDesafiado($desafiado) {$this->desafiado = $desafiado;}
    function getrounds() {return $this->rounds;}
    function setrounds($rounds) {$this->rounds = $rounds;}
    function getAprovada() {return $this->aprovada;}
    function setAprovada($aprovada) {$this->aprovada = $aprovada;}
} 



